mozci/task.py

```diff
@@ -262,33 +262,54 @@ def _load_errorsummary(self):
             return
 
         groups = None
+        group_results = {}
 
         lines = [json.loads(l) for l in self.get_artifact(path).splitlines()]
+
+        has_group_result = any(line["action"] == "group_result" for line in lines)
+
         for line in lines:
             if line["action"] == "test_groups":
                 groups = list(set(line["groups"]) - {"default"})
 
-            elif line["action"] == "test_result":
+            # After enough time has passed since https://bugzilla.mozilla.org/show_bug.cgi?id=1632242,
+            # we can switch to look for `group_result` lines exclusively.
+            elif not has_group_result and line["action"] == "test_result":
                 group = line.get("group")
                 if group == "default":
                     continue
 
-                self._results.append(
-                    GroupResult(group=group, ok=line["status"] == line["expected"],)
-                )
+                # The "OK" case should never happen given how errorsummary.log works, but
+                # better to be safe than sorry.
+                if line["expected"] == line["status"]:
+                    if group not in group_results:
+                        group_results[group] = "OK"
+                else:
+                    group_results[group] = "ERROR"
+
+            elif line["action"] == "group_result":
+                group_results[line["group"]] = line["status"]
 
             elif line["action"] == "log":
                 self._errors.append(line["message"])
 
+        self._results = [
+            GroupResult(group, True if result == "OK" else False)
+            for group, result in group_results.items()
+            if result != "SKIP"
+        ]
+
         if groups is not None:
             # Assume all groups for which we have no results passed.
-            # In practice, they could also have been skipped (we should ignore them when
-            # it will be feasible, https://github.com/mozilla/mozci/issues/166).
-            known_results = {result.group for result in self._results}
+            # After enough time has passed since https://bugzilla.mozilla.org/show_bug.cgi?id=1632242
+            # and https://bugzilla.mozilla.org/show_bug.cgi?id=1631515, we
+            # can stop assuming since all groups will have an associated result.
```

Comment:
Ditto, just specify how much time is "enough" directly in the comment.

Reply:
Good idea, I've done that also for another case in the same file (I haven't done it here in particular because one of the two bugs isn't fixed yet).